src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -862,6 +862,15 @@ async def get_requirements(
     direct_deps_tgts = await MultiGet(
         Get(Targets, DependenciesRequest(tgt.get(Dependencies))) for tgt in owned_by_us
     )
+    direct_deps_chained = OrderedSet(itertools.chain.from_iterable(direct_deps_tgts))
+    # If a python_requirement T has an undeclared requirement R, we recommend fixing that by adding
+    # an explicit dependency from T to a python_requirement target for R. In that case we want to
+    # represent these explicit deps in T's distribution metadata. See issue #17593.
+    transitive_3rd_party_tgts = await MultiGet(
```

Comment:
```suggestion
    transitive_explicit_reqs = await MultiGet(
```

Reply:
I think we should be clear that these are *explicit* requirements, and not the actual full set of transitive requirements.

Also "3rd party" terminology is possibly confusing, since we use the term "requirements" elsewhere for this.